src/transformers/dynamic_module_utils.py

```diff
@@ -154,7 +154,17 @@ def get_class_in_module(class_name, module_path):
         shutil.copy(f"{module_dir}/{module_file_name}", tmp_dir)
         # On Windows, we need this character `r` before the path argument of `os.remove`
         cmd = f'import os; os.remove(r"{module_dir}{os.path.sep}{module_file_name}")'
-        subprocess.run(["python", "-c", cmd])
+        # We don't know which python binary file exists in an environment. For example, if `python3` exists but not
+        # `python`, the call `subprocess.run(["python", ...])` gives `FileNotFoundError` (about python binary). Notice
+        # that, if the file to be removed is not found, we also have `FileNotFoundError`, but it is not raised to the
+        # caller's process.
+        try:
+            subprocess.run(["python", "-c", cmd])
+        except FileNotFoundError:
+            try:
+                subprocess.run(["python3", "-c", cmd])
+            except FileNotFoundError:
+                pass
```

Comment:
This `removing file and copy back` trick is necessary to make dynamic module loading **always** work (it's not just about testing). However, the fix is hacky, and with this `python`/`python3` checks, it's too far from the original purpose of this method `get_class_in_module`.

I can wrap this hacky logic of removing file temporarily in a method if you think it's better too.